modules/statistics/CallStats.js

```diff
@@ -65,167 +61,483 @@ const reportType = {
+            const originalMethod = theBackend[methodName];
 
-/**
- * Checks whether we need to re-initialize callstats and starts the process.
- * @private
- */
-CallStats._checkInitialize = function() {
-    if (CallStats.initialized || !CallStats.initializeFailed
-        || !callStats || CallStats.initializeInProgress) {
-        return;
+            theBackend[methodName] = function(...theArguments) {
+                logger.debug(methodName, theArguments);
+                originalMethod.apply(theBackend, theArguments);
+            };
+        }
+        const originalReportError = theBackend.reportError;
+
+        /* eslint-disable max-params */
+        theBackend.reportError
+        = function(pc, cs, type, ...otherArguments) {
+            // Logs from the logger are submitted on the applicationLog event
+            // "type". Logging the arguments on the logger will create endless
+            // loop, because it will put all the logs to the logger queue again.
+            if (type === wrtcFuncNames.applicationLog) {
+                // NOTE otherArguments are not logged to the console on purpose
+                // to not log the whole log batch
+                // FIXME check the current logging level (currently not exposed
+                // by the logger implementation)
+                console && console.debug('reportError', pc, cs, type);
+            } else {
+                logger.debug('reportError', pc, cs, type, ...otherArguments);
+            }
+            try {
+                originalReportError.call(
+                    theBackend, pc, cs, type, ...otherArguments);
+            } catch (exception) {
+                if (type === wrtcFuncNames.applicationLog) {
+                    console && console.error('reportError', exception);
+                } else {
+                    GlobalOnErrorHandler.callErrorHandler(exception);
+                }
+            }
+        };
+
+        /* eslint-enable max-params */
     }
 
-    // callstats object created, not initialized and it had previously failed,
-    // and there is no init in progress, so lets try initialize it again
-    CallStats.initializeInProgress = true;
-    callStats.initialize(
-        callStats.callStatsID,
-        callStats.callStatsSecret,
-        callStats.userID,
-        initCallback.bind(callStats));
-};
+    /**
+     * Initializes the CallStats backend. Should be called only if
+     * {@link CallStats.isBackendInitialized} returns <tt>false</tt>.
+     * @param {object} options
+     * @param {String} options.callStatsID CallStats credentials - ID
+     * @param {String} options.callStatsSecret CallStats credentials - secret
+     * @param {string} options.userName the <tt>userName</tt> part of
+     * the <tt>userID</tt> aka display name, see CallStats docs for more info.
+     * @param {string} options.aliasName the <tt>aliasName</tt> part of
+     * the <tt>userID</tt> aka endpoint ID, see CallStats docs for more info.
```

Comment:
It may seem, so but according to the CallStats dashboard it's this misleading way around. I'll double check to be sure.